doc/source/whatsnew/v0.20.0.txt

```diff
@@ -705,6 +706,69 @@ New Behavior:
    TypeError: Cannot compare 2014-01-01 00:00:00 of
    type <class 'pandas.tslib.Timestamp'> to string column
 
+.. _whatsnew_0200.api_breaking.index_order:
+
+Index order after inner join due to Index intersection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``Index.intersection`` now preserves the order of the calling Index (left)
+instead of the other Index (right) (:issue:`15582`). This affects the inner
+joins (methods ``DataFrame.join`` and ``pd.merge``) and the .align methods.
+
+- ``Index.intersection``
+
+  .. ipython:: python
+
+     idx1 = pd.Index([2, 1, 0])
+     idx1
+     idx2 = pd.Index([1, 2, 3])
+     idx2
+
+  Previous Behavior:
+
+  .. code-block:: ipython
+
+     In [4]: idx1.intersection(idx2)
+     Out[4]: Int64Index([1, 2], dtype='int64')
+
+  New Behavior:
+
+  .. ipython:: python
+
+     idx1.intersection(idx2)
+
+- ``DataFrame.join`` and ``pd.merge``
+
+  .. ipython:: python
+
+     df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+     df1
```

Comment:
No, the inner join on columns is not affected. Indeed it was already working as expected, preserving the order of the left column.